octoprint_LEDStripControl/__init__.py

```diff
@@ -19,26 +19,77 @@
 import re
 
 import octoprint.plugin
-import RPi.GPIO as GPIO
+import pigpio
+try:
+	import RPi.GPIO as GPIO
+except (ImportError, RuntimeError):
+	GPIO = None
+
+phy_to_bcm= { 0:None, 1:None, 2:None, 3:2, 4:None, 5:3, 6:None, 7:4, 8:14,
+			 9:None, 10:15, 11:17, 12:18, 13:27, 14:None, 15:22, 16:23,
+			 17:None, 18:24, 19:10, 20:None, 21:9, 22:25, 23:11, 24:8, 25:None,
+			 26:7, 27:0, 28:1, 29:5, 30:None, 31:6, 32:12, 33:13, 34:None,
+			 35:19, 36:16, 37:26, 38:20, 39:None, 40:21 }
+
+class PiGPIOpin(object):
+	def __init__(self, pigpiod, pin, logger):
+		self._pigpiod = pigpiod
+		self._logger = logger
+
+		# attempt to convert the physical pin to a bcm pin
+		# how is this not in a library already?
+		if type(phy_to_bcm[pin]) is int:
```

Comment:
if phy_to_bcm[pin] is not None:
Do you care if pin is invalid?  Raises KeyError instead of GIGO.